zebrad/src/commands/start/sync.rs

```diff
@@ -174,22 +176,20 @@ where
                         // It indicates that the remote peer does not have any blocks
                         // following the prospective tip.
                         // TODO(jlusby): reject both main and test net genesis blocks
-                        if hashes[0] == super::GENESIS {
-                            tracing::debug!("skipping response that does not extend the tip");
-                            continue;
+                        match hashes.first() {
+                            Some(&super::GENESIS) => {
+                                tracing::debug!("skipping response that does not extend the tip");
+                                continue;
+                            }
+                            Some(_) | None => {}
```

Comment:
this seems more opaque to me than the previous logic. i'm not sure, but i think it may not be equivalent: if the response is `[genesis]`, than the previous logic will reject it (because the tip is extracted after validating the response) but the new logic won't.  i think it would be better to keep the previous logic, and just change it to keep the tip separate.